src/Projection/AbstractReadModelProjection.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Projection;
+
+use Prooph\EventStore\EventStore;
+use Prooph\EventStore\Exception\RuntimeException;
+use Prooph\EventStore\StreamName;
+
+abstract class AbstractReadModelProjection extends AbstractProjection
+{
+    /**
+     * @var ReadModelProjection
+     */
+    private $readModelProjection;
+
+    public function __construct(EventStore $eventStore, string $name, ReadModelProjection $readModelProjection)
+    {
+        parent::__construct($eventStore, $name, false);
+
+        $this->readModelProjection = $readModelProjection;
+    }
+
+    public function readModelProjection(): ReadModelProjection
+    {
+        return $this->readModelProjection;
+    }
+
+    public function delete(bool $deleteEmittedEvents): void
```

Comment:
This is part of the interface and will delete the projection. As the in memory adapter is not persistent, there is nothing more to do here. But for other event store implementations things are different.